frontend/src/core/user/selectors.js

```diff
@@ -0,0 +1,53 @@
+/* @flow */
+
+import { createSelector } from 'reselect';
+
+import * as navigation from 'core/navigation';
+
+import { NAME } from '.';
+
+import type { NavigationParams } from 'core/navigation';
+import type { UserState } from 'core/user';
+
+
+const userSelector = (state): UserState => state[NAME];
+
+
+export function _isTranslator(
+    user: UserState,
+    parameters: NavigationParams
+): boolean {
+    const locale = parameters.locale;
+    const project = parameters.project;
+    const localeProject = locale + "-" + project;
+
+    if (!user.isAuthenticated) {
+        return false;
+    }
+
+    if (user.managerForLocales.indexOf(locale) !== -1) {
+        return true;
+    }
+
+    if (user.translatorForProjects.hasOwnProperty(localeProject)) {
+        return user.translatorForProjects[localeProject];
```

Comment:
I'm getting a flow error here that I can't fix:
`Cannot get user.translatorForProjects[localeProject] because string [1] is not an array index.`

Reply:
`user.translatorForProjects` is defined as an `Array<string>` in `core/user/reducer.js`. Hence this code shouldn't work, because that array is not supposed to have your `localeProject` property, nor can it be accessed like an object. I'm not sure which one is wrong, your code or the type declaration, but you'll have to change one of the 2.